### ISSUE TITLE
Unwanted behaviors in the latest version 7.42, which does not occur in version 7.40. 

### ISSUE THREAD

**Issue by sonhsien** [tonikelope/megabasterd#337]
Dear author,
First, thank you for the nice program helping a lot for intensive mega downloads.
Just a few info I would like to comment, regarding the latest version 7.42.

1) The downloads got stuck at 99% and 100% quite often.
2) The joining file chunks process take a lot longer.
3) When the above occurs, my OS got kind of kid-napped or having troubling responding.

These do not occur in the old version, e.g., 7.40 (of course, with the same settings). So, it maybe helpful by comparing these two versions.
I hope these issues can be solved in the future version.

Thank you again for your contribution.
Ps, if you think this is the issue to be merged, please do so.

Best.

**Comment by tonikelope** [comment]
Yes, there is something weird. I just reproduced it. As soon as I have some time I will check it. THANKS FOR THE WARNING

**Comment by tonikelope** [comment]
I hope the latest version works correctly for you. If not, reopen the issue and paste a debug file and test links.
